bot/moderation/moderation.py

```diff
@@ -282,15 +829,194 @@ async def _send_confirmation_dialog(self, ctx: commands.Context, embed: discord.
         await message.add_reaction(constants.EMOJI_CANCEL)
 
         def check_reaction(_reaction, user):
-            return _reaction.message.id == ctx.message.id and user == ctx.author and \
+            return user == ctx.author and _reaction.message.id == message.id and \
                    str(_reaction.emoji) in [constants.EMOJI_CANCEL, constants.EMOJI_CONFIRM]
 
         reaction = await self.bot.wait_for('reaction_add', timeout=constants.TIMEOUT_USER_SELECTION,
                                            check=check_reaction)
         await message.delete()
 
+        if str(reaction[0].emoji) == constants.EMOJI_CANCEL:
+            await ctx.message.delete()
+
         return str(reaction[0].emoji) == constants.EMOJI_CONFIRM
 
+    @commands.Cog.listener(name='on_member_update')
+    @commands.Cog.listener(name='on_user_update')
+    async def name_change(self, before: discord.Member, after: discord.Member):
+        """Event listener which triggers if a member has changed his name.
+
+        The old name (or serverwide nickname) will be saved along with the user id and a timestamp in our database.
+        This way we can have an overview about who had which name on the server during a specific timespan.
+
+        Args:
+            before (discord.Member): The old member object before the update.
+            after (discord.Member): The new member object after the update.
+        """
+        if before.display_name != after.display_name:
+            self._db_connector.add_member_name(before.id, before.display_name, datetime.utcnow())
+
+
+async def _scheduled_unmute_user(server_id: int, ch_rules_id: int, role_id: int, user_id: int):
+    """Method which is being called by the scheduler if the specified amount of time for the corresponding tempmute has
+    ran out.
+
+    Unmutes a user with the specified ID on a specific server.
+
+    Args:
+        server_id (int): The id of the server where the user should be unbanned from.
+        ch_rules_id (int): The id of the rules channel.
+        role_id (int): The id of the mute role which should be removed from the user.
```

Comment:
If the role is fixed (always unmute) then why do you pass it as an argument?

Reply:
If I try to get the ID via the constants file, MyPy gets angry because it couldn't be in there and is therefore an `Optional`.